pkg/providers/gcp/provider_postgres.go

```diff
@@ -503,10 +512,10 @@ func (p *PostgresProvider) buildCloudSQLCreateStrategy(ctx context.Context, pg *
 	return nil
 }
 
-func buildDefaultCloudSQLSecret(p *v1alpha1.Postgres) *v1.Secret {
+func buildDefaultCloudSQLSecret(p *v1alpha1.Postgres) (*v1.Secret, error) {
 	password, err := resources.GeneratePassword()
 	if err != nil {
-		return nil
+		return nil, nil
```

Comment:
Returning a nil error will cause any caller of `buildDefaultCloudSQLSecret()` to assume everything is fine with the call.
Example:
```
secret, err := buildDefaultCloudSQLSecret(pg)
if err != nil {
  // err is always nil, even if there was an error
}
```

Reply:
Updated to return error. Sorry I missed this!